addons/account/models/account_move.py

```diff
@@ -310,216 +222,374 @@ def _get_default_invoice_incoterm(self):
-    # Technical field to hide Reconciled Entries stat button
     has_reconciled_entries = fields.Boolean(compute="_compute_has_reconciled_entries")
     show_reset_to_draft_button = fields.Boolean(compute='_compute_show_reset_to_draft_button')
-    # Credit Limit related field
     partner_credit_warning = fields.Text(
         compute='_compute_partner_credit_warning',
-        groups="account.group_account_invoice,account.group_account_readonly")
-
-    # ==== Hash Fields ====
-    restrict_mode_hash_table = fields.Boolean(related='journal_id.restrict_mode_hash_table')
-    secure_sequence_number = fields.Integer(string="Inalteralbility No Gap Sequence #", readonly=True, copy=False)
-    inalterable_hash = fields.Char(string="Inalterability Hash", readonly=True, copy=False)
-    string_to_hash = fields.Char(compute='_compute_string_to_hash', readonly=True)
+        groups="account.group_account_invoice,account.group_account_readonly",
+    )
 
-    # We neeed the btree index for unicity constraint (on field) AND this one for human searches
     def _auto_init(self):
-        super(AccountMove, self)._auto_init()
+        super()._auto_init()
         if sql.install_pg_trgm(self._cr):
+            # We need the btree index for unicity constraint (on field) AND this one for human searches
             sql.create_index(self._cr, 'account_move_name_trigram_index', self._table, ['"name" gin_trgm_ops'], 'gin')
 
-    @api.model
-    def _field_will_change(self, record, vals, field_name):
-        if field_name not in vals:
-            return False
-        field = record._fields[field_name]
-        if field.type == 'many2one':
-            return record[field_name].id != vals[field_name]
-        if field.type == 'many2many':
-            current_ids = set(record[field_name].ids)
-            after_write_ids = set(record.new({field_name: vals[field_name]})[field_name].ids)
-            return current_ids != after_write_ids
-        if field.type == 'one2many':
-            return True
-        if field.type == 'monetary' and record[field.get_currency_field(record)]:
-            return not record[field.get_currency_field(record)].is_zero(record[field_name] - vals[field_name])
-        if field.type == 'float':
-            record_value = field.convert_to_cache(record[field_name], record)
-            to_write_value = field.convert_to_cache(vals[field_name], record)
-            return record_value != to_write_value
-        return record[field_name] != vals[field_name]
-
-    @api.model
-    def _cleanup_write_orm_values(self, record, vals):
-        cleaned_vals = dict(vals)
-        for field_name, value in vals.items():
-            if not self._field_will_change(record, vals, field_name):
-                del cleaned_vals[field_name]
-        return cleaned_vals
+        self.env.cr.execute("""
+            CREATE INDEX IF NOT EXISTS account_move_to_check_idx
+            ON account_move(journal_id) WHERE to_check = true;
```

Comment:
Why not a composed index and why is it that important on the to_check?  (I would think an index just on journal_id should do)

Reply:
I didn't change those lines, just moved it.
But to answer your question, this index was added to improve the performance of the dashboard (bank/cash tiles) for one specific query that took a lot of time compared to the rest and that was often run